core/control-plane/control-plane-aggregate-services/src/main/java/org/eclipse/edc/connector/controlplane/services/transferprocess/TransferProcessProtocolServiceImpl.java

```diff
@@ -232,12 +232,22 @@ private ServiceResult<TransferProcess> suspendedAction(TransferSuspensionMessage
 
     @NotNull
     private ServiceResult<TransferProcess> terminatedAction(TransferTerminationMessage message, TransferProcess transferProcess) {
+        if (transferProcess.getType() == PROVIDER) {
+            var termination = dataFlowManager.terminate(transferProcess);
+            if (termination.failed()) {
+                return ServiceResult.conflict("Cannot terminate transfer process %s: %s".formatted(transferProcess.getId(), termination.getFailureDetail()));
+            }
+        }
```

Comment:
I guess this took inspiration from the `suspendedAction` but looking at this now makes me wonder: what if the transferProcess cannot be terminated? (see condition at line 241), in that case the flow will be stopped but the state won't change.
I think this logic should stay in the if branch after line 241, maybe that if could be come a guard, it will make everything more readable